LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,337 @@
+            if self.isEditing {
+                // TODO: confirm
+                self.delegate?.save()
+            }
+            self.isEditing.toggle()
+        }
+        return Button( action: action ) {
+            Group {
+                if isEditing {
+                    Text(NSLocalizedString("Done", comment: "Done button text"))
+                } else {
+                    Text(NSLocalizedString("Edit", comment: "Edit button text"))
+                }
+            }
+            .onTapGesture {
+                action()
+            }
+            .onLongPressGesture {
+                self.editButtonDesignVersion = self.editButtonDesignVersion.cycle()
+                AudioServicesPlayAlertSound(kSystemSoundID_Vibrate)
+            }
+        }
+    }
+    
+    private var correctionRangeSection: some View {
+        SectionWithEdit(isEditing: $isEditing,
+                        editButtonDesignVersion: $editButtonDesignVersion,
+                        title: NSLocalizedString("Correction Range", comment: "Correction Range section title"),
+                        footer: EmptyView(),
+                        gotoEdit: { self.delegate?.gotoEdit(therapySetting: TherapySetting.glucoseTargetRange) })
+        {
+            ForEach(self.viewModel.therapySettings.glucoseTargetRangeSchedule?.items ?? [], id: \.self) { value in
+                ScheduledRangeItem(time: value.startTime, range: value.value, unit: self.unit, guardrail: Guardrail.correctionRange)
+            }
+        }
+    }
+    
+    private var temporaryCorrectionRangesSection: some View {
+        SectionWithEdit(isEditing: $isEditing,
+                        editButtonDesignVersion: $editButtonDesignVersion,
+                        title: NSLocalizedString("Temporary Correction Ranges", comment: "Temporary Correction Ranges section title"),
+                        footer: EmptyView(),
+                        gotoEdit: { self.delegate?.gotoEdit(therapySetting: TherapySetting.correctionRangeOverrides) })
+        {
+            Group {
+                if self.viewModel.therapySettings.glucoseTargetRangeSchedule != nil {
+                    ForEach([ CorrectionRangeOverrides.Preset.preMeal, CorrectionRangeOverrides.Preset.workout  ], id: \.self) { preset in
+                        CorrectionRangeOverridesRangeItem(
+                            preMealTargetRange: self.viewModel.therapySettings.preMealTargetRange,
+                            workoutTargetRange: self.viewModel.therapySettings.workoutTargetRange,
+                            unit: self.unit,
+                            preset: preset,
+                            correctionRangeScheduleRange: self.viewModel.therapySettings.glucoseTargetRangeSchedule!.scheduleRange()
+                        )
+                    }
+                } else {
+                    EmptyView()
+                }
+            }
+        }
+    }
+    
+}
+
+struct ScheduledRangeItem: View {
```

Comment:
[nit, let me know if you just want overall feedback!] To follow the other naming convention, maybe make this `ScheduleRangeItem`?

Reply:
All feedback is welcome!